src/Subscriptions/Models/Subscription.php

```diff
@@ -109,15 +110,28 @@ public function donations(): ModelQueryBuilder
         return give()->donations->queryBySubscriptionId($this->id);
     }
 
+    /**
+     * @unreleased
+     *
+     * @return ModelQueryBuilder<SubscriptionNote>
+     */
+    public function notes(): ModelQueryBuilder
+    {
+        return give()->subscriptions->notes->queryBySubscriptionId($this->id);
+    }
+
     /**
      * Get Subscription notes
      *
+     * @deprecated Access notes via $subscription->notes()->getAll() instead.
      * @since 2.19.6
      *
      * @return object[]
      */
     public function getNotes(): array
     {
+        _give_deprecated_function(__METHOD__, '4.6.0', '$subscription->notes()->getAll()');
+
         return give()->subscriptions->getNotesBySubscriptionId($this->id);
```

Comment:
Can we use `$this->notes()->getAll()` instead of the legacy method?

Reply:
I did not add that at this time because of the custom data being currently returned in `getNotesBySubscriptionId`. Replacing the content here now would break things in case someone uses this method. Important to note that internally, it's not being used or at least I could not find where we are using it.